src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\PhpUnit;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Michał Adamski <michal.adamski@gmail.com>
+ */
+final class PhpUnitShortWillReturnFixer extends AbstractFixer
+{
+    /**
+     * @internal
+     */
+    const RETURN_METHODS_MAP = [
+        'returnValue' => 'willReturn',
+        'returnSelf' => 'willReturnSelf',
+        'returnValueMap' => 'willReturnMap',
+        'returnArgument' => 'willReturnArgument',
+        'returnCallback' => 'willReturnCallback',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Usage of `->will($this->returnValue(..))` statements must be replaced by its shorter equivalent `->willReturn(...)`.',
+            [
+                new CodeSample('<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testSomeTest()
+    {
+        $someMock = $this->createMock(Some::class);
+        $someMock->method("some")->will($this->returnSelf());
+        $someMock->method("some")->will($this->returnValue("example"));
+        $someMock->method("some")->will($this->returnArgument(2));
+        $someMock->method("some")->will($this->returnCallback("str_rot13"));
+        $someMock->method("some")->will($this->returnValueMap(["a","b","c"]));
+    }
+}
+'),
+            ],
+            null,
+            'Risky when PHPUnit classes are overridden or not accessible, or when project has PHPUnit incompatibilities.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_OBJECT_OPERATOR, T_STRING]);
```

Comment:
`T_CLASS` as well I think and using `PhpUnitTestCaseIndicator::findPhpUnitClasses`? Or do we want to fix these cases everywhere?

Reply:
Yes, but step-by-step, I'm waiting for Travis and then will push `PhpUnitTestCaseIndicator` with test fix - you can see some tests have invalid PHP syntax. Then I will add more test cases, like uppercases names, `willDoSomething` or `will` being property and fix them as well and then ask for review ;)